src/core/lib/slice/slice.h

```diff
@@ -249,7 +248,11 @@ class MutableSlice : public slice_detail::BaseSlice,
       : slice_detail::BaseSlice(slice) {
     GPR_DEBUG_ASSERT(slice.refcount == nullptr || slice.refcount->IsUnique());
   }
-  ~MutableSlice() { grpc_slice_unref_internal(c_slice()); }
+  ~MutableSlice() {
+    if (reinterpret_cast<uintptr_t>(c_slice().refcount) > 1) {
```

Comment:
I officially do not understand compilers at all. :)  I would have expected that this would *add* bloat, not remove it, since we're duplicating this existing code.

Where does the bloat reduction come from here?

Reply:
Mostly the copying in of the slice to the ref/unref functions contribute a fairly high number of instructions versus what happens in the ref/unref.
If `grpc_slice_ref` took a `grpc_slice*` we'd have a different story... but that's in the realms of a rather large change.